arrays/arrays.c

```diff
@@ -114,33 +137,57 @@ void arr_append(Array *arr, char *element) {
  *
  * Throw an error if the value is not found.
  *****/
-void arr_remove(Array *arr, char *element) {
-
-  // Search for the first occurence of the element and remove it.
-  // Don't forget to free its memory!
-
-  // Shift over every element after the removed element to the left one position
+void arr_remove(Array *arr, char *element)
+{
+  int elementfound = 0;
+  int i = 0;
+  while (i < arr->count)
+  {
+    // Search for the first occurence of the element and remove it.
+    // Don't forget to free its memory!
+    if (elementfound == 0 && arr->elements[i] == element)
```

Comment:
Comparing string elements with `==` is...not the worse thing you can do, but it's a dangerous game. `==` only compares the pointer values of the strings in question, not the strings themselves. So consider a case like this:
char *string1 = "hello";
char *string2 = "hello";

There's a reasonable chance that these could trigger a truthy value when comparing with `==`. The OS might do this:

`printf("%p\n", string1); //0x400634`
`printf("%p\n", string2); //0x400634`

You'll see that the pointers values are the same even though the variable names are different. Just a quirk of the OS to be either efficient or lazy depending on how you think about it. 

But consider if you did:
`char *string1 = "hello";`
`char string2[] = "hello";`

You might get:
`printf("%p\n", string1); //0x400634`
`printf("%p\n", string2); //0x7ffc73824b7a`

So if you compare these with `==` you'll actually get a falsy value. What you want to do instead is use the strcmp() method which returns 0 if the two strings are the same. It actually compares the strings rather than comparing the pointer value. 

So the TL;DR version is use strcmp() to compare strings, don't use `==`

Reply:
Thank you! Makes sense.